### PR TITLE
Map Editor separate WAM file.

### PR DESCRIPTION
**Pull request by Waclaw-I**
We now use WAM (Workadventure Map File) file format to store everything map editor related.

This PR includes:

- Separating Tiled Areas objects from ones created by map editor
- WAM file format definition
- Creating WAM file on map upload
- Load/Save areas / entities / lastCommandId within WAM file
- Proper handling of WAM file across entire app

### CODE REVIEW

File: libs/map-editor/src/GameMap/GameMapAreas.ts
Comment (moufmouf) [inline libs/map-editor/src/GameMap/GameMapAreas.ts:1]: I have the feeling that `updateAreaByName` from `GameMapAreas` is only used by `updateAreaByName` from `GameMapFrontWrapper` and that this `GameMapFrontWrapper.updateAreaByName` is not used at all.

Maybe we can remove those?

Comment (moufmouf) [reply libs/map-editor/src/GameMap/GameMapAreas.ts:1]: Same thing for `GameMapFrontWrapper.deleteAreaByName`

My real question: when do we need to access an area by name? Can't we always fall back to using the ID instead?

File: play/src/front/Phaser/Game/GameScene.ts [hunk 1]
```diff
@@ -378,16 +384,25 @@ export class GameScene extends DirtyScene {
...
+                    undefined,
+                    undefined,
+                    (key: string, type: string, wamFile: unknown) => {
+                        this.wamFile = WAMFileFormat.parse(wamFile);
+                        this.mapUrlFile = new URL(this.wamFile.mapUrl, this.wamUrlFile).toString();
+                        this.doLoadTMJFile(this.mapUrlFile);
+                    }
+                )
+                .catch((e) => {
+                    console.error(e);
```
[the remark this answers is not in the document]
Comment (Waclaw-I) [reply play/src/front/Phaser/Game/GameScene.ts:402]: obraz
Not sure if this is considered to be enough.

[the remark this answers is not in the document]
Comment (moufmouf) [reply play/src/front/Phaser/Game/GameScene.ts:402]: Excellent :+1:

**Pull request merged**